halyard-deploy/src/main/java/com/netflix/spinnaker/halyard/deploy/config/v1/secrets/DecryptingObjectMapper.java

```diff
@@ -0,0 +1,151 @@
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.netflix.spinnaker.halyard.deploy.config.v1.secrets;
+
+import com.fasterxml.jackson.annotation.JsonInclude;
+import com.fasterxml.jackson.core.JsonGenerator;
+import com.fasterxml.jackson.databind.*;
+import com.fasterxml.jackson.databind.module.SimpleModule;
+import com.fasterxml.jackson.databind.ser.BeanPropertyWriter;
+import com.fasterxml.jackson.databind.ser.BeanSerializerModifier;
+import com.fasterxml.jackson.databind.ser.std.StdScalarSerializer;
+import com.netflix.spinnaker.config.secrets.EncryptedSecret;
+import com.netflix.spinnaker.halyard.config.config.v1.secrets.SecretSessionManager;
+import com.netflix.spinnaker.halyard.config.model.v1.node.Secret;
+import com.netflix.spinnaker.halyard.config.model.v1.node.SecretFile;
+import com.netflix.spinnaker.halyard.deploy.spinnaker.v1.profile.Profile;
+import org.apache.commons.lang.RandomStringUtils;
+
+import java.io.IOException;
+import java.lang.reflect.Field;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.Iterator;
+import java.util.List;
+
+/**
+ * DecryptingObjectMapper serializes (part of) Halyard configurations, decrypting secrets contained in fields
+ * annotated with @Secret.
+ *
+ * It also decrypts the content of secret files, assign them a random name, and store them
+ * in {@link Profile} to be serialized later.
+ *
+ * decryptedOutputDirectory is the path to the decrypted secret files on the service's host.
+ */
+public class DecryptingObjectMapper extends ObjectMapper {
+    private enum SecretType {
+        SECRET_FILE, SECRET, NO_SECRET
+    }
+
+    protected Profile profile;
+    protected Path decryptedOutputDirectory;
+
+    public DecryptingObjectMapper(SecretSessionManager secretSessionManager, Profile profile, Path decryptedOutputDirectory) {
+        super();
+        this.configure(DeserializationFeature.FAIL_ON_UNKNOWN_PROPERTIES, true);
+        this.setSerializationInclusion(JsonInclude.Include.NON_NULL);
+
+        this.profile = profile;
+        this.decryptedOutputDirectory = decryptedOutputDirectory;
+
+        SimpleModule module = new SimpleModule();
+        module.setSerializerModifier(new BeanSerializerModifier() {
+
+            @Override
+            public List<BeanPropertyWriter> changeProperties(SerializationConfig config, BeanDescription beanDesc, List<BeanPropertyWriter> beanProperties) {
+                Class _class = beanDesc.getBeanClass();
+
+                for (Iterator<BeanPropertyWriter> it = beanProperties.iterator(); it.hasNext(); ) {
```

Comment:
I think the preferred way to write this is

`for (BeanPropertyWriter it : beanProperties) {`